e2e-playwright/dashboards-suite/dashboard-adhoc-filters.spec.ts

```diff
@@ -0,0 +1,114 @@
+import { Page, Locator } from '@playwright/test';
+
+import { test, expect } from '@grafana/plugin-e2e';
+
+import testDashboard from '../dashboards/AdHocFilterTest.json';
+
+import jsonData from './dashboard-adhoch-filters-data.json';
+
+// Helper function to get a specific cell in a table
+const getCell = async (loc: Page | Locator, rowIdx: number, colIdx: number) =>
+  loc
+    .getByRole('row')
+    .nth(rowIdx)
+    .getByRole(rowIdx === 0 ? 'columnheader' : 'gridcell')
+    .nth(colIdx);
+
+test.use({
+  featureToggles: {
+    dashboardDsAdHocFiltering: true,
+    adhocFiltersInTooltips: true,
+    tableNextGen: true,
+    kubernetesDashboards: process.env.KUBERNETES_DASHBOARDS === 'true',
+  },
+});
+
+test.describe(
+  'Dashboard with adhoc filters',
+  {
+    tag: ['@dashboards'],
+  },
+  () => {
+    let dashboardUID: string;
+
+    test.beforeAll(async ({ request }) => {
+      // Import the test dashboard
+      const response = await request.post('/api/dashboards/import', {
+        data: {
+          dashboard: testDashboard,
+          folderUid: '',
+          overwrite: true,
+          inputs: [],
+        },
+      });
+      const responseBody = await response.json();
+      dashboardUID = responseBody.uid;
+    });
+
+    test.afterAll(async ({ request }) => {
+      // Clean up the imported dashboard
+      if (dashboardUID) {
+        await request.delete(`/api/dashboards/uid/${dashboardUID}`);
+      }
+    });
+
+    test('Should show adhoc filters', async ({ page, gotoDashboardPage, selectors }) => {
+      // Handle query and query_range API calls
+      await page.route(/\/api\/ds\/query/, async (route) => {
+        await route.fulfill({
+          status: 200,
+          contentType: 'application/json',
+          body: JSON.stringify(jsonData),
+        });
+      });
+
+      const dashboardPage = await gotoDashboardPage({ uid: dashboardUID });
+
+      const panel = dashboardPage.getByGrafanaSelector(selectors.components.Panels.Panel.title('New panel'));
```

Comment:
I feel like `'New panel'` is the kind of thing that's not meaningful, and so could break. This is another place we could rename potentially, if we felt like it.

```suggestion
      const panel = dashboardPage.getByGrafanaSelector(selectors.components.Panels.Panel.title('Table powered by Prometheus'));
```